app/views/annotations/create.js.erb

```diff
@@ -23,7 +26,14 @@ resultComponent.addAnnotation({
     start: <%= [@annotation.y1, @annotation.y2].min || 'undefined' %>,
     end: <%= [@annotation.y1, @annotation.y2].max || 'undefined' %>
   }
-});
+},
+<%= @annotation.annotation_text.annotation_category.flexible_criterion_id || 'undefined' %>,
+<%= @annotation.result.marks.find_by(markable_id: @annotation.annotation_text&.annotation_category.flexible_criterion_id,
```

Comment:
Because it's a polymorphic association, you can do `find_by(markable: <criterion object>)` rather than specifying the `_id` and `_type` fields separately.